packages/dm-core-plugins/src/generic-list/GenericTablePlugin.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { ChangeEvent, useContext, useEffect, useState } from 'react'
+import {
+  AuthContext,
+  DmssAPI,
+  ErrorResponse,
+  IUIPlugin,
+  Loading,
+  TGenericObject,
+  useDocument,
+} from '@development-framework/dm-core'
+import { AxiosError, AxiosResponse } from 'axios'
+import { Button, Icon, Input, Progress, Table } from '@equinor/eds-core-react'
+import { delete_to_trash, library_add } from '@equinor/eds-icons'
+
+type TGenericTablePlugin = {
+  editMode: boolean
+  columns: string[]
+  showDelete: boolean
+}
+const defaultConfig: TGenericTablePlugin = {
+  editMode: true,
+  columns: ['name', 'type'],
+  showDelete: true,
+}
+
+export const GenericTablePlugin = (
+  props: IUIPlugin & { config?: TGenericTablePlugin }
+) => {
+  const { idReference, config } = props
+  const internalConfig = { ...defaultConfig, ...config }
+  const [document, loading, _, error] = useDocument<TGenericObject[]>(
+    idReference,
+    2
+  )
+  const [items, setItems] = useState<{ [key: string]: TGenericObject }>({})
+  const [isSaveLoading, setIsSaveLoading] = useState<boolean>(false)
+  const { token } = useContext(AuthContext)
+  const dmssAPI = new DmssAPI(token)
+
+  useEffect(() => {
+    if (loading || !document) return
+    // Need to generate a uuid for each item in the list to be used for reacts "key" property
+    const itemsWithIds = Object.fromEntries(
+      Object.values(document).map((value: any) => [crypto.randomUUID(), value])
+    )
+    setItems(itemsWithIds)
+  }, [document, loading])
+
+  const deleteItem = (reference: string, key: string) => {
+    dmssAPI.documentRemove({ idReference: reference }).then(() => {
+      delete items[key]
+      setItems({ ...items })
+    })
+  }
+  const addItem = (reference: string, type: string) => {
+    dmssAPI
+      // TODO: Get type from parent blueprint, be able to select specialised type
+      .instantiateEntity({
+        entity: { type: items[Object.keys(items)[0]].type },
+      })
+      .then((newEntity: AxiosResponse<object, TGenericObject>) => {
+        dmssAPI
+          .documentAdd({ absoluteRef: reference, body: newEntity.data })
+          .then(() =>
+            setItems({
+              ...items,
+              [crypto.randomUUID()]: { ...newEntity.data },
+            })
+          )
+      })
+  }
+
+  const updateItem = (
+    key: string,
+    attribute: string,
+    newValue: string | number | boolean
+  ) => {
+    items[key][attribute] = newValue
+    setItems({ ...items })
+  }
+
+  const saveTable = () => {
```

Comment:
I find this functionality a bit confusing. If a table has a save button, than as a user I would expect it to be needed also when deleting or adding item. Let me know if you want to have a discussing on this

Reply:
Restructured logic to only update database on save button click.